src/loc2d_ros.cpp

```diff
@@ -140,6 +148,21 @@ lama::Loc2DROS::~Loc2DROS()
     delete tfb_;
 }
 
+void lama::Loc2DROS::publishCurrentPose(bool init)
+{
+    if (init)
+    {
+        current_pose_ = loc2d_.getPose();
+        current_orientation_ = tf::createQuaternionFromYaw(current_pose_.rotation());
+    }
```

Comment:
Why the need for the `init`? This function should always get the current pose from the `loc2d_` object.

Reply:
Right, with other updates I mentioned in comments, I deleted this.